tests/datatypes/test_str.py

```diff
@@ -778,6 +778,17 @@ def test_splitlines(self):
         print(s1.splitlines(True))
         """)
 
+    def test_zfill(self):
+        self.assertCodeExecution("""
+            s = '42'
+            print(s.zfill(5))
+        """)
+
+    def test_zfill_negative(self):
+        self.assertCodeExecution("""
+            s = '-42'
+            print(s.zfill(5))
```

Comment:
This test is not doing anything of value, the "negative" thing here is a string and not a number, the `zfill` method is getting the same argument as in the other test.

Reply:
Sorry I dont follow. `zfill` just checks if the first char is a `-` or a `+`, and then inserts it at the first, it doesn't matter if its a number or a string.

Reply:
Ooops, my apologies, you're right!
Indeed, it seems the missing tests are for the empty string, a positive sign and something like `'-.2'.zfill(5)`

Reply:
Might also want to include test cases from CPython https://github.com/python/cpython/blob/master/Lib/test/string_tests.py#L830

Reply:
Wow almost got that bug in without the empty string test case, :) I have updated the tests.

Reply:
Cool, thanks, this is getting good! :+1: 
May I ask you to group these happy-path tests in the same assertion in same test case?
While in many projects it's good to have separate cases, in VOC each separate assertion requires compiling the code given as argument and run it separately in Java and CPython -- so putting several related cases under the same test saves some resources.